### PR TITLE
Added symlink int test and fix for Cluster Master CRD

### PR DESCRIPTION
**Pull request by pdhanoya-splunk**
- Ported Arjun's fix for Cluster Manager to Cluster Master [PR 1053](https://github.com/splunk/splunk-operator/pull/1053) 
- Added new test to check new index through smartstore for Cluster master and cluster manager CR's

### CODE REVIEW

File: test/smartstore/manager_smartstore_test.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ var _ = Describe("Smartstore test", func() {
 		}
 	})
 
-	Context("Standalone Deployment (S1)", func() {
+	XContext("Standalone Deployment (S1)", func() {
```
Comment (akondur) [inline test/smartstore/manager_smartstore_test.go:44]: Did we want to disable this test?

File: test/smartstore/manager_smartstore_test.go [hunk 2]
```diff
@@ -92,7 +92,7 @@ var _ = Describe("Smartstore test", func() {
 		})
 	})
 
-	Context("Standalone Deployment (S1)", func() {
+	XContext("Standalone Deployment (S1)", func() {
```
Comment (akondur) [inline test/smartstore/manager_smartstore_test.go:95]: Same comment as above

File: pkg/splunk/enterprise/util.go [hunk 3]
```diff
@@ -650,7 +650,7 @@ var resetSymbolicLinks = func(ctx context.Context, client splcommon.ControllerCl
 
 	// Create command for symbolic link creation
 	var command string
-	if crKind == "ClusterManager" {
+	if crKind == "ClusterManager" || crKind == "ClusterMaster" {
 		command = setSymbolicLinkCmanager
```
Comment (akondur) [inline pkg/splunk/enterprise/util.go:654]: I believe for clusterMaster the location is etc/master-apps while the string points to [setSymbolicLinkCmanager](https://github.com/splunk/splunk-operator/blob/1deadcb2641c9c6d945222963acfcf26e351f436/pkg/splunk/enterprise/names.go#L132). Has the soft link been tested? Maybe I am missing something here.

Comment (pdhanoya-splunk) [reply pkg/splunk/enterprise/util.go:654]: The location is same for Cluster Master as well in any 9.x.x splunk.

**Review by akondur — APPROVED** [review]

File: pkg/splunk/enterprise/clustermaster.go [hunk 4]
```diff
@@ -195,9 +200,11 @@ func ApplyClusterMaster(ctx context.Context, client splcommon.ControllerClient,
 			}
 		}
 
+		// Create podExecClient
+		podExecClient := splutil.GetPodExecClient(client, cr, "")
+
```
Comment (gaurav-splunk) [inline pkg/splunk/enterprise/clustermaster.go:205]: any specific reason for moving it out?

Comment (pdhanoya-splunk) [reply pkg/splunk/enterprise/clustermaster.go:205]: This is direct port of fix from Arjun here : https://github.com/splunk/splunk-operator/pull/1053

Comment (akondur) [reply pkg/splunk/enterprise/clustermaster.go:205]: @gaurav-splunk @pdhanoya-splunk This change is not required. It was a part of an earlier logic workflow which was changed later due to a review comment. IMO to maintain symmetry we can keep it as is for now.

**Review by vivekr-splunk — APPROVED** [review]

**Review by gaurav-splunk — APPROVED** [review]

**Pull request merged**